### PR TITLE
fix(batch-exports): Fix double encoding of properties in Snowflake

### PR DESCRIPTION
**Pull request by hazzadous**
Previously we were simply writing the JSON representation of the
properties to the Snowflake table. This meant that the properties
were double encoded, and the user would have to decode them
themselves.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/temporal/workflows/snowflake_batch_export.py [hunk 1]
```diff
@@ -144,8 +144,17 @@ async def insert_into_snowflake_activity(inputs: SnowflakeInsertInputs):
                     if not result:
                         break
 
+                    # Make sure the properties attribute is parsed as JSON
+                    # TODO: this may well be a change that is worth making in
+                    # the other batch export workflows as well, although I leave
+                    # this as a followup.
```
Comment (tomasfarias) [inline posthog/temporal/workflows/snowflake_batch_export.py:150]: I think this is necessary change, but I don't like that *we* are handling it. The ChClient should return a dict object already. Unless the CH column is typed as String?

Comment (tomasfarias) [reply posthog/temporal/workflows/snowflake_batch_export.py:150]: Can the `ChClient` handle this? Saves us from having to worry about in every single export.

Comment (tomasfarias) [reply posthog/temporal/workflows/snowflake_batch_export.py:150]: Also, this makes it independent of Temporal and the batch export implementation 🤔...

Comment (hazzadous) [reply posthog/temporal/workflows/snowflake_batch_export.py:150]: Yeah I don't think there is a way in ClickHouse to have the type be JSON. And I don't _think_ there is any way for the client to be instructed to convert to a dict.

I think a better implementation than what we have now is to implement this within the iterator returned from get_results_iterator for the fields that should be unmarshalled, but I was planning on doing that as a follow-up when also handling this in the S3 export, what do you think?

**Review by tomasfarias — APPROVED** [review]
We'll do the generic "fix" once we confirm this is safe.

**Pull request merged**